src/Mod/PartDesign/App/FeatureChamfer.cpp

```diff
@@ -144,16 +155,57 @@ App::DocumentObjectExecReturn *Chamfer::execute()
 
     this->positionByBaseFeature();
 
+#ifdef FC_USE_TNP_FIX
+    if ( static_cast<Part::ChamferType>(chamferType) == Part::ChamferType::distanceAngle ) {
+        size2 = angle;
+    }
+#else
     //If no element is selected, then we use a copy of previous feature.
     if (SubNames.empty()) {
         this->Shape.setValue(TopShape);
         return App::DocumentObject::StdReturn;
     }
-
     // create an untransformed copy of the basefeature shape
     Part::TopoShape baseShape(TopShape);
     baseShape.setTransform(Base::Matrix4D());
+#endif
     try {
+#ifdef FC_USE_TNP_FIX
+        TopoShape shape(0);
+        shape.makeElementChamfer(TopShape,
+                                 edges,
+                                 static_cast<Part::ChamferType>(chamferType),
+                                 size,
+                                 size2,
+                                 nullptr,
+                                 flipDirection ? Part::Flip::flip : Part::Flip::none);
+        if (shape.isNull()) {
+            return new App::DocumentObjectExecReturn(
+                QT_TRANSLATE_NOOP("Exception", "Failed to create chamfer"));
+        }
+
+        TopTools_ListOfShape aLarg;
+        aLarg.Append(TopShape.getShape());
+        bool failed = false;
+        if (!BRepAlgo::IsValid(aLarg, shape.getShape(), Standard_False, Standard_False)) {
+            ShapeFix_ShapeTolerance aSFT;
+            aSFT.LimitTolerance(shape.getShape(),
+                                Precision::Confusion(),
+                                Precision::Confusion(),
+                                TopAbs_SHAPE);
+        }
+        if (!failed) {
+            shape = refineShapeIfActive(shape);
+            shape = getSolid(shape);
+        }
+        this->Shape.setValue(shape);
+        if (failed) {
+            return new App::DocumentObjectExecReturn(
+                QT_TRANSLATE_NOOP("Exception", "Resulting shape is invalid"));
+        }

```

Comment:
@bgbsww 
`failed` is never set true.
Does the LimitTolerance has any effect? `Precision::Confusion` is the default value if I'm not mistaken.

Reply:
@FlachyJoe unfortunately bgbsww will not be able to respond to your question: https://blog.freecad.org/2024/10/08/rip-bradley-bgbsww-mclean/

Can you check LinkStage3 and see if this code matches what's there?

Reply:
I'm so sad to learn this :disappointed: